src/common/SurgeLogger.cpp

```diff
@@ -0,0 +1,76 @@
+#include "SurgeLogger.h"
+#include "spdlog/sinks/basic_file_sink.h"
+#include "spdlog/sinks/stdout_sinks.h"
+
+#include "spdlog/logger.h"
+#include "spdlog/async.h"
+#include <memory>
+#include <filesystem.h>
+
+/*
+** This file exists to have a single global static which is initialized at library open time
+** thereby making sure that the log is set up as we want when it runs. 
+**
+** Two parts. The "init" function below sets up the logger in the OS reasonable way.
+** For now it is stderr+~/Library/Logs on macos and stderr only on windows and linux.
+**
+** The second part is the static variable which is initialized with the function call
+*/
+
+namespace SurgeLogInternal
+{
+
+typedef std::shared_ptr< spdlog::logger > log_t;
+namespace fs = std::experimental::filesystem;
+
+log_t internalSurgeLogInit()
+{
+    std::vector<spdlog::sink_ptr> sinks;
+    
+    /*
+    ** Where to put the logfile is actually OS dependent. 
+    ** The approach we take here is to add a pair of sinks
+    ** to the logger, the first to stderr
+    */
+    auto stderrLogger = std::make_shared<spdlog::sinks::stderr_sink_mt>();
+    sinks.push_back( stderrLogger );
+    
+    /*
+    ** And if someone has implemented this for their OS, the second to a standard
+    ** logfile location
+    */
+
+
+#if MAC 
+    char lfb[ PATH_MAX ];
+    sprintf(lfb, "%s/Library/Logs/Surge.log", getenv("HOME"));
+    auto fileSink = make_shared<spdlog::sinks::basic_file_sink_mt> (lfb);
```

Comment:
Should this be opt-in instead of always on? Is the log always appended to - is there some auto-truncation size? What happens when multiple processes might be accessing the same log? Seems to me logging to the console by default is much preferred and opting in to dump to a log if necessary is a nicer default behavior. Just my 2c.

Reply:
If I had any idea at all what happened to stderr or stdout ejected from a plugin in logic, au lab, and hosting au, I would be fine to skip this. Do you know?

That said: SPD has an auto rotating log. I could use that. Or I could comment this section out and just leave it on in my development area inside an #ifdef log-to-file or some such.

Reply:
@baconpaul can you run Logic from the command line? If I run Reaper, Live, and Bitwig from command line, I get all the `stdout` dumped to the terminal; don't have Logic to see if it's the same, but I assume so?

Reply:
(an aside: I usually set up REAPER as the app to run in XCode with a project that just loads the plugin and opens it by default, in this case, the terminal output is just dumped to the XCode console)